diff --git a/framework/src/interface/function/function.h b/framework/src/interface/function/function.h
index ad8319819..f6fb65ac6 100644
--- a/framework/src/interface/function/function.h
+++ b/framework/src/interface/function/function.h
@@ -817,6 +817,8 @@ public:
 
     const std::unordered_set<std::string> &LoopIdxNameList() { return loopIdxNameList_; }
     bool InsertLoopIdxNameList(const std::string &idxName);
+    void SetCurrentPassIndex(size_t passIndex) {currentPassIndex_ = passIndex;}
+    int GetCurrentPassIndex() const {return currentPassIndex_;}
 private:
     int functionMagic_{-1};
     std::string funcMagicName_; // Function name
@@ -830,6 +832,7 @@ private:
     std::vector<std::string> calleeMagicNameList_;
     std::unordered_set<std::string> loopIdxNameList_;
     bool isUnderDynamicFunction_{false};
+    int currentPassIndex_{-1};
 
     std::vector<std::shared_ptr<LogicalTensor>> originInCasts_;
     std::unordered_set<std::shared_ptr<LogicalTensor>> inCastsSet_; // Input tensors set
diff --git a/framework/src/interface/interpreter/calc.h b/framework/src/interface/interpreter/calc.h
index 6c3d9616c..953a00789 100644
--- a/framework/src/interface/interpreter/calc.h
+++ b/framework/src/interface/interpreter/calc.h
@@ -192,6 +192,9 @@ inline void ScatterUpdate(LogicalTensorDataPtr out, LogicalTensorDataPtr self, L
     int axis = -2, std::string cacheMode = "BSND", int blockSize = 1) {
     GetCalcOps()->ScatterUpdate(out, self, index, axis, cacheMode, blockSize);
 }
+inline void LogicalView(LogicalTensorDataPtr out, LogicalTensorDataPtr self, Offset offset) {
+    GetCalcOps()->LogicalView(out, self, offset);
+}
 inline void Scatter(LogicalTensorDataPtr out, LogicalTensorDataPtr self, LogicalTensorDataPtr index, const Element &src,
     int axis, int reduce) {
     GetCalcOps()->Scatter(out, self, index, src, axis, reduce);
diff --git a/framework/src/interface/interpreter/calc_api.h b/framework/src/interface/interpreter/calc_api.h
index 83938915e..8a3fbd2b3 100644
--- a/framework/src/interface/interpreter/calc_api.h
+++ b/framework/src/interface/interpreter/calc_api.h
@@ -99,6 +99,7 @@ struct CalcOps {
     void (*ReduceAcc)(LogicalTensorDataPtr, const std::vector<LogicalTensorDataPtr> &);
     void (*Copy)(LogicalTensorDataPtr, LogicalTensorDataPtr, bool);
     void (*ScatterUpdate)(LogicalTensorDataPtr, LogicalTensorDataPtr, LogicalTensorDataPtr, int, std::string, int);
+    void (*LogicalView)(LogicalTensorDataPtr, LogicalTensorDataPtr, Offset);
     void (*Scatter)(LogicalTensorDataPtr, LogicalTensorDataPtr, LogicalTensorDataPtr, const Element &, int, int);
     void (*FormatND2NZ)(LogicalTensorDataPtr, LogicalTensorDataPtr);
     void (*FormatNZ2ND)(LogicalTensorDataPtr, LogicalTensorDataPtr);
diff --git a/framework/src/interface/interpreter/calc_common.cpp b/framework/src/interface/interpreter/calc_common.cpp
index 6be7412a8..b84474e8b 100644
--- a/framework/src/interface/interpreter/calc_common.cpp
+++ b/framework/src/interface/interpreter/calc_common.cpp
@@ -42,7 +42,21 @@ REGISTER_CALC_OP(OP_SYNC_DST, Opcode::OP_SYNC_DST, ExecuteOpNone);
 REGISTER_CALC_OP(OP_BAR_V, Opcode::OP_BAR_V, ExecuteOpNone);
 REGISTER_CALC_OP(OP_BAR_M, Opcode::OP_BAR_M, ExecuteOpNone);
 REGISTER_CALC_OP(OP_NOP, Opcode::OP_NOP, ExecuteOpNone);
-REGISTER_CALC_OP(OP_VIEW, Opcode::OP_VIEW, ExecuteOpNone); // View作为视图操作，操作前后是同一块内存
+
+void ExecuteOpView(ExecuteOperationContext *ctx) {
+    ASSERT(ctx->ooperandInplaceDataViewList->size() == 1);
+    ASSERT(ctx->ioperandDataViewList->size() == 1);
+    auto &oop = ctx->ooperandInplaceDataViewList->at(0);
+    auto iop = ctx->ioperandDataViewList->at(0);
+    auto opAttr = std::static_pointer_cast<ViewOpAttribute>(ctx->op->GetOpAttribute());
+    auto offset = ctx->opInter->EvaluateOffset(opAttr->GetFromOffset(), opAttr->GetFromDynOffset());
+    if (oop->GetData() == iop->GetData()) {
+        return;
+    }
+    calc::LogicalView(oop, iop, offset);
+
+}
+REGISTER_CALC_OP(OP_VIEW, Opcode::OP_VIEW, ExecuteOpView); 
 
 void ExecuteOpCopyOut(ExecuteOperationContext *ctx) {
     ASSERT(ctx->ooperandInplaceDataViewList->size() == 1);
diff --git a/framework/src/interface/interpreter/calculator/calc_torch.cpp b/framework/src/interface/interpreter/calculator/calc_torch.cpp
index eca0437a2..10b59c8fd 100644
--- a/framework/src/interface/interpreter/calculator/calc_torch.cpp
+++ b/framework/src/interface/interpreter/calculator/calc_torch.cpp
@@ -90,6 +90,13 @@ static torch::Tensor View(const torch::Tensor &self, const std::vector<int64_t>
     return self.as_strided(shape, self.strides(), storageOffset);
 }
 
+static void LogicalView(LogicalTensorDataPtr out, LogicalTensorDataPtr self, Offset offset) {
+    auto shape = out->GetShape();
+    auto tself = From(self);
+    auto view = View(tself, shape, offset);
+    From(out) = view;
+}
+
 static bool AllClose(LogicalTensorDataPtr self, LogicalTensorDataPtr other, double atol, double rtol) {
     return From(self).allclose(From(other), atol, rtol);
 }
@@ -861,6 +868,7 @@ static struct CalcOps calcOps = {
     .ReduceAcc = ReduceAcc,
     .Copy = Copy,
     .ScatterUpdate = ScatterUpdate,
+    .LogicalView = LogicalView,
     .Scatter = Scatter,
     .FormatND2NZ = FormatND2NZ,
     .FormatNZ2ND = FormatNZ2ND,
diff --git a/framework/src/interface/interpreter/function.h b/framework/src/interface/interpreter/function.h
index e71b64f01..a9c900e76 100644
--- a/framework/src/interface/interpreter/function.h
+++ b/framework/src/interface/interpreter/function.h
@@ -26,6 +26,7 @@ namespace npu::tile_fwk {
 struct FunctionIODataPair {
     std::vector<std::shared_ptr<LogicalTensorData>> incastDataViewList;
     std::vector<std::shared_ptr<LogicalTensorData>> outcastDataViewList;
+    std::shared_ptr<FunctionIODataPair> rootInoutDataPair;
 
     FunctionIODataPair() {}
     FunctionIODataPair(std::vector<std::shared_ptr<LogicalTensorData>> incastDataViewList_,
@@ -451,10 +452,11 @@ struct FunctionInterpreter {
         return AllocateDataView(frame, tensor, tensor->GetRawTensor()->GetDataType(), inplaceTensor);
     }
 
-    void ExecuteOpCallLeaf(ExecuteOperationContext *ctx) {
+    void ExecuteOpCallLeaf(ExecuteOperationContext *ctx, std::shared_ptr<FunctionIODataPair> &rootInoutDataPair) {
         Function *callee = GetCallee(ctx->op);
         auto inoutDataPair =
             std::make_shared<FunctionIODataPair>(*ctx->ioperandDataViewList, *ctx->ooperandInplaceDataViewList);
+        inoutDataPair->rootInoutDataPair = rootInoutDataPair;
         ExecuteFunctionFrame(callee, ctx->op, inoutDataPair);
     }
 
@@ -481,7 +483,8 @@ struct FunctionInterpreter {
 
     void ExecuteInplaceOperation(FunctionFrame &frame, Operation &op, int oOperandIdx,
         const std::vector<std::shared_ptr<LogicalTensorData>> &iOpDataList,
-        std::vector<std::shared_ptr<LogicalTensorData>> &oOpDataList) {
+        std::vector<std::shared_ptr<LogicalTensorData>> &oOpDataList,
+        std::shared_ptr<FunctionIODataPair> &inoutDataPair) {
         auto oop = op.GetOOperands()[oOperandIdx];
         auto index = GetInplaceIndex(&op, oOperandIdx);
         ASSERT(index != -1);
@@ -493,11 +496,34 @@ struct FunctionInterpreter {
             Offset viewOffsets = EvaluateOffset(opAttr->GetFromOffset(), opAttr->GetFromDynOffset());
             auto validShape = EvaluateValidShape(oop->GetDynValidShape());
             auto rawShape = EvaluateValidShape(oop->GetRawTensor()->GetDynRawShape());
-            auto ret = frame.AllocateDataView(
-                oop, viewOffsets, validShape, rawShape, oop->GetRawTensor()->GetDataType(), iop);
+            std::shared_ptr<LogicalTensorData> ret;
+            if (frame.func->GetCurrentPassIndex() > 4) {
+                ret = frame.AllocateDataView(oop, viewOffsets, validShape, rawShape, oop->GetRawTensor()->GetDataType(), iop);
+            } else {
+                ret = AllocateDataView(frame, oop);
+            }
             oOpDataList.emplace_back(ret);
         } else {
             oOpDataList.emplace_back(AllocateDataView(frame, oop, iop));
+            auto it = std::find(frame.func->outCasts_.begin(), frame.func->outCasts_.end(), oop);
+            if (it != frame.func->outCasts_.end()) {
+                ASSERT(frame.tensorDataViewDict.count(oop) != 0);
+                auto oopDataView = frame.tensorDataViewDict[oop]; 
+                ASSERT(frame.tensorDataViewDict.count(iop) != 0);
+                auto newPtr = frame.tensorDataViewDict[iop]; 
+                auto targetPair = inoutDataPair->rootInoutDataPair ? 
+                              inoutDataPair->rootInoutDataPair : 
+                              inoutDataPair;
+                bool updated = false;
+                for (auto& ptr : targetPair->outcastDataViewList) {
+                    if (ptr.get() == oopDataView.get()) {
+                        ptr = newPtr;
+                        updated = true;
+                        break;
+                    }
+                }      
+            ASSERT(updated); 
+            }
         }
     }
 
@@ -511,7 +537,7 @@ struct FunctionInterpreter {
         return false;
     }
 
-    void ExecuteOperation(FunctionFrame &frame, Operation *op) {
+    void ExecuteOperation(FunctionFrame &frame, Operation *op, std::shared_ptr<FunctionIODataPair> &inoutDataPair) {
         auto iOpDataList = frame.GetDataViewList(op->GetIOperands());
         for (size_t index = 0; index < iOpDataList.size(); index++) {
             if (iOpDataList[index] == nullptr) {
@@ -524,7 +550,7 @@ struct FunctionInterpreter {
         for (size_t i = 0; i < op->GetOOperands().size(); i++) {
             auto oop = op->GetOOperands()[i];
             if (auto index = GetInplaceIndex(op, i); index != -1) {
-                ExecuteInplaceOperation(frame, *op, i, iOpDataList, oOpDataList);
+                ExecuteInplaceOperation(frame, *op, i, iOpDataList, oOpDataList, inoutDataPair);
             } else {
                 if (isConsumerAccMatmul(op)) {
                     auto dtype = oop->GetRawTensor()->GetDataType();
@@ -534,14 +560,15 @@ struct FunctionInterpreter {
                     }
                     oOpDataList.push_back(AllocateDataView(frame, oop, dtype));
                 } else {
-                    oOpDataList.push_back(AllocateDataView(frame, oop));
+                    auto ret = AllocateDataView(frame, oop);
+                    oOpDataList.push_back(ret);
                 }
             }
         }
         ExecuteOperationContext ctx = {&frame, {}, op, &iOpDataList, {}, &oOpDataList};
 
         if (op->GetOpcode() == Opcode::OP_CALL) {
-            ExecuteOpCallLeaf(&ctx);
+            ExecuteOpCallLeaf(&ctx, inoutDataPair);
         } else {
             TimeStamp ts;
             operationInterpreter->ExecuteOperation(&ctx);
@@ -601,7 +628,7 @@ struct FunctionInterpreter {
             if (op.GetOpcode() == Opcode::OP_PRINT && verifyType != VerifyType::TENSOR_GRAPH)
                 continue;
             ExecuteHandleOperationBegin(&op);
-            ExecuteOperation(*frame, &op);
+            ExecuteOperation(*frame, &op, inoutDataPair);
             ExecuteHandleOperationEnd();
         }
         ExecuteHandleFunctionEnd();
diff --git a/framework/src/interface/interpreter/operation.h b/framework/src/interface/interpreter/operation.h
index 55d4e7c85..b08b05f6e 100644
--- a/framework/src/interface/interpreter/operation.h
+++ b/framework/src/interface/interpreter/operation.h
@@ -88,7 +88,7 @@ private:
             if (validShape == dataView->GetShape()) {
                 result.emplace_back(dataView);
             } else {
-                result.emplace_back(dataView->View(validShape, std::vector<int64_t>(validShape.size(), 0)));
+                result.emplace_back(dataView->View(validShape, dataView->GetOffset()));
             }
         }
         return result;
diff --git a/framework/src/passes/pass_mgr/pass_manager.cpp b/framework/src/passes/pass_mgr/pass_manager.cpp
index cc5d51059..a439d86b5 100644
--- a/framework/src/passes/pass_mgr/pass_manager.cpp
+++ b/framework/src/passes/pass_mgr/pass_manager.cpp
@@ -236,6 +236,7 @@ Status PassManager::RunPass(Program &program, Function &function, const std::str
             passDfxCfg.dumpGraph = true;
         }
         pass->SetPassConfigs(passDfxCfg);
+        function.SetCurrentPassIndex(i);
         ALOG_INFO_F("[PassManager] Apply pass <%s> on function: %s.", identifier.c_str(), function.GetMagicName().c_str());
         auto start = std::chrono::high_resolution_clock::now();
         if (pass->Run(function, strategy, identifier, i) != SUCCESS) {
