diff --git a/framework/src/codegen/cloudnpu/codegen_vector.cpp b/framework/src/codegen/cloudnpu/codegen_vector.cpp
index ca3c55de42b2deaff21afa9f67792846d6ff958c..0c17a633de730a0ed71692838b4148c1d0c7d94f 100644
--- a/framework/src/codegen/cloudnpu/codegen_vector.cpp
+++ b/framework/src/codegen/cloudnpu/codegen_vector.cpp
@@ -721,13 +721,12 @@ std::string CodeGenOpCloudNPU::PrintCumSumDynamicUnaligned(const PrintCumSumPara
     // template params
     std::vector<std::string> paramList;
     paramList.insert(paramList.end(), {dataTypeExpr[ID0]});
-    for (size_t i = 1; i < inputRawShape.size(); ++i) {
+    for (size_t i = 0; i < inputRawShape.size(); ++i) {
         paramList.emplace_back(std::to_string(inputRawShape[i]));
     }
 
-    int axis = param.axis + SHAPE_DIM4 - param.inputRawShape.size(); // 调用4维tileop需要切换axis
     bool flag = param.flag;
-    paramList.emplace_back(std::to_string(axis));
+    paramList.emplace_back(std::to_string(param.axis));
     paramList.emplace_back(std::to_string(flag));
     std::string templateParam = JoinString(paramList, ", ");
 
@@ -769,6 +768,7 @@ std::string CodeGenOpCloudNPU::GenCumSumOp() const {
 
     ASSERT(opAttrs.count(OP_ATTR_PREFIX + "axis")) << "cannot get axis attr";
     int axis = npu::tile_fwk::AnyCast<int64_t>(opAttrs.at(OP_ATTR_PREFIX + "axis"));
+    axis = axis + SHAPE_DIM4 - inputRawShape.size(); // 调用4维tileop需要切换axis
 
     ASSERT(opAttrs.count(OP_ATTR_PREFIX + "flag")) << "cannot get flag attr";
     bool flag = npu::tile_fwk::AnyCast<bool>(opAttrs.at(OP_ATTR_PREFIX + "flag"));
diff --git a/framework/src/interface/operation/opcode.cpp b/framework/src/interface/operation/opcode.cpp
index a69c3973999b83f3715c910f42571bdda1690098..01afeeed2b54afb1ec3290ba4633982ccd07c559 100644
--- a/framework/src/interface/operation/opcode.cpp
+++ b/framework/src/interface/operation/opcode.cpp
@@ -162,7 +162,7 @@ OpcodeManager::OpcodeManager() {
     registerInfo(Opcode::OP_SCATTER_SCALAR, OpCoreType::ANY, "SCATTER_SCALAR",
         {MemoryType::MEM_UB, MemoryType::MEM_UB}, {MemoryType::MEM_UB}, {}, OpCalcType::OTHER);
     registerInfo(Opcode::OP_CUM_SUM, OpCoreType::AIV, "CUM_SUM", {MemoryType::MEM_UB}, {MemoryType::MEM_UB},
-        {"TileOp::TcumSum", PIPE_V, PIPE_V, CoreType::AIV}, OpCalcType::OTHER, {OP_ATTR_PREFIX + "axis", OP_ATTR_PREFIX + "flag"});
+        {"TileOp::TcumSum", PIPE_V, PIPE_V, CoreType::AIV}, OpCalcType::OTHER, {OP_ATTR_PREFIX + "axis", OP_ATTR_PREFIX + "flag", OpAttributeKey::excludeBufferReuse});
     registerInfo(Opcode::OP_MAXIMUM, OpCoreType::AIV, "MAXIMUM", {MemoryType::MEM_UB, MemoryType::MEM_UB}, {MemoryType::MEM_UB}, {"TileOp::Tmax", PIPE_V, PIPE_V, CoreType::AIV}, OpCalcType::BROADCAST, {OpAttributeKey::inputCombineAxis}, TileShapeVerifier::Verify);
     registerInfo(Opcode::OP_MINIMUM, OpCoreType::AIV, "MINIMUM", {MemoryType::MEM_UB, MemoryType::MEM_UB}, {MemoryType::MEM_UB}, {"TileOp::Tmin", PIPE_V, PIPE_V, CoreType::AIV}, OpCalcType::BROADCAST, {OpAttributeKey::inputCombineAxis}, TileShapeVerifier::Verify);
     registerInfo(Opcode::OP_PAIRMAX, OpCoreType::AIV, "PAIRMAX", {MemoryType::MEM_UB, MemoryType::MEM_UB}, {MemoryType::MEM_UB}, {"TileOp::Tmaxpair", PIPE_V, PIPE_V, CoreType::AIV}, OpCalcType::BROADCAST, {OpAttributeKey::inputCombineAxis, OpAttributeKey::excludeBufferReuse});
diff --git a/framework/src/interface/operation/vector/math.cpp b/framework/src/interface/operation/vector/math.cpp
index ba94c4a5841a04a080293595213bf2e5b1adfd72..342c44be88beeadfd555f98db57d013e25a91dde 100644
--- a/framework/src/interface/operation/vector/math.cpp
+++ b/framework/src/interface/operation/vector/math.cpp
@@ -397,35 +397,39 @@ void InnerTiledCumSum(size_t cur, Function &function, const TileShape &tileShape
     const LogicalTensorPtr &dstTensor = cumSumPara.dstTensor;
     const int axis = cumSumPara.axis;
     const bool flag = cumSumPara.flag;
+    auto &vecTile = tileShape.GetVecTile();
 
     if (cur == dstTensor->shape.size()) {
         auto dstTile = dstTensor->View(function, cumSumTileInfo.dstTileInfo.shape, cumSumTileInfo.dstTileInfo.offset);
         auto inputTile = input->View(function, cumSumTileInfo.inputTileInfo.shape, cumSumTileInfo.inputTileInfo.offset);
-        auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {inputTile}, {dstTile});
+
+        LogicalTensorPtr srcTile = std::make_shared<LogicalTensor>(function, dstTile->Datatype(), dstTile->GetShape());
+        auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {inputTile}, {srcTile});
         op.SetAttribute(OP_ATTR_PREFIX + "axis", axis);
         op.SetAttribute(OP_ATTR_PREFIX + "flag", flag);
+
+        std::vector<int64_t> offset = cumSumTileInfo.dstTileInfo.offset;
+        if (offset[axis] > 0) {
+            offset[axis] -= 1;
+            std::vector<int64_t> shape = cumSumTileInfo.dstTileInfo.shape;
+            shape[axis] = 1;
+            LogicalTensorPtr lastAxisTile = dstTensor->View(function, shape, offset);
+            LogicalTensorPtr lastTile = std::make_shared<LogicalTensor>(function, srcTile->Datatype(), srcTile->GetShape());
+            auto &eop = function.AddOperation("TILE_EXPAND", {lastAxisTile}, {lastTile});
+            eop.SetAttribute(OP_ATTR_PREFIX + "EXPANDDIM", axis);
+            function.AddOperation(Opcode::OP_ADD, {srcTile, lastTile}, {dstTile});
+        } else {
+            function.AddOperation(Opcode::OP_REGISTER_COPY, {srcTile}, {dstTile});
+        }
         return;
     }
-
-    auto &vecTile = tileShape.GetVecTile();
     int64_t tmpTile = vecTile[cur];
 
-    if (static_cast<int>(cur) == axis) {
-        tmpTile = input->GetShape()[cur];
-    }
-
     for (int i = 0; i < input->GetShape()[cur]; i += tmpTile) {
-        if (static_cast<int>(cur) == axis) {
-            cumSumTileInfo.dstTileInfo.offset[cur] = 0;
-            cumSumTileInfo.dstTileInfo.shape[cur] = dstTensor->shape[cur];
-            cumSumTileInfo.inputTileInfo.offset[cur] = 0;
-            cumSumTileInfo.inputTileInfo.shape[cur] = input->shape[cur];
-        } else {
-            cumSumTileInfo.dstTileInfo.offset[cur] = i;
-            cumSumTileInfo.dstTileInfo.shape[cur] = std::min(input->shape[cur] - i, tmpTile);
-            cumSumTileInfo.inputTileInfo.offset[cur] = i;
-            cumSumTileInfo.inputTileInfo.shape[cur] = std::min(input->shape[cur] - i, tmpTile);
-        }
+        cumSumTileInfo.dstTileInfo.offset[cur] = i;
+        cumSumTileInfo.dstTileInfo.shape[cur] = std::min(input->shape[cur] - i, tmpTile);
+        cumSumTileInfo.inputTileInfo.offset[cur] = i;
+        cumSumTileInfo.inputTileInfo.shape[cur] = std::min(input->shape[cur] - i, tmpTile);
         InnerTiledCumSum(cur + 1, function, tileShape, cumSumPara, cumSumTileInfo);
     }
 }
@@ -436,15 +440,40 @@ void TiledCumSum(Function &function, const TileShape &tileShape, const CumSumPar
     CumSumTileInfoPara cumSumTileInfo{
         TileInfo(cumSumPara.Input->GetShape().size(), cumSumPara.Input->GetOffset().size()),
         TileInfo(cumSumPara.dstTensor->GetShape().size(), cumSumPara.dstTensor->GetOffset().size())};
+
     InnerTiledCumSum(0, function, tileShape, cumSumPara, cumSumTileInfo);
     return;
 }
 
 void TensorCumSum(Function &function, const CumSumPara &cumSumPara) {
-    auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {cumSumPara.Input}, {cumSumPara.dstTensor});
-    op.SetAttribute(OP_ATTR_PREFIX + "axis", cumSumPara.axis);
-    op.SetAttribute(OP_ATTR_PREFIX + "flag", cumSumPara.flag);
-    return;
+    if (cumSumPara.Input->Datatype() == DT_INT8) {
+        LogicalTensorPtr inputConverted = std::make_shared<LogicalTensor>(function, DT_FP16, cumSumPara.Input->GetShape());
+        Operation &castInputOp = function.AddOperation(Opcode::OP_CAST, {cumSumPara.Input}, {inputConverted});
+        castInputOp.SetAttribute(OP_ATTR_PREFIX + "mode", CastMode::CAST_NONE);
+        LogicalTensorPtr dstConverted = std::make_shared<LogicalTensor>(function, DT_FP16, cumSumPara.dstTensor->GetShape());
+        auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {inputConverted}, {dstConverted});
+        op.SetAttribute(OP_ATTR_PREFIX + "axis", cumSumPara.axis);
+        op.SetAttribute(OP_ATTR_PREFIX + "flag", cumSumPara.flag);
+        Operation &castDstOp = function.AddOperation(Opcode::OP_CAST, {dstConverted}, {cumSumPara.dstTensor});
+        castDstOp.SetAttribute(OP_ATTR_PREFIX + "mode", CastMode::CAST_NONE);
+        return;
+    } else if (cumSumPara.Input->Datatype() == DT_BF16 || cumSumPara.Input->Datatype() == DT_FP16) {
+        LogicalTensorPtr inputConverted = std::make_shared<LogicalTensor>(function, DT_FP32, cumSumPara.Input->GetShape());
+        Operation &castInputOp = function.AddOperation(Opcode::OP_CAST, {cumSumPara.Input}, {inputConverted});
+        castInputOp.SetAttribute(OP_ATTR_PREFIX + "mode", CastMode::CAST_NONE);
+        LogicalTensorPtr dstConverted = std::make_shared<LogicalTensor>(function, DT_FP32, cumSumPara.dstTensor->GetShape());
+        auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {inputConverted}, {dstConverted});
+        op.SetAttribute(OP_ATTR_PREFIX + "axis", cumSumPara.axis);
+        op.SetAttribute(OP_ATTR_PREFIX + "flag", cumSumPara.flag);
+        Operation &castDstOp = function.AddOperation(Opcode::OP_CAST, {dstConverted}, {cumSumPara.dstTensor});
+        castDstOp.SetAttribute(OP_ATTR_PREFIX + "mode", CastMode::CAST_NONE);
+        return;
+    } else {
+        auto &op = function.AddOperation(Opcode::OP_CUM_SUM, {cumSumPara.Input}, {cumSumPara.dstTensor});
+        op.SetAttribute(OP_ATTR_PREFIX + "axis", cumSumPara.axis);
+        op.SetAttribute(OP_ATTR_PREFIX + "flag", cumSumPara.flag);
+        return;
+    }
 }
 
 Tensor CumSum(const Tensor &input, const int &axis) {
@@ -452,21 +481,44 @@ Tensor CumSum(const Tensor &input, const int &axis) {
     auto shapeSize = input.GetShape().size();
     auto dataType = input.GetDataType();
 
-    ASSERT(SHAPE_DIM1 <= shapeSize && shapeSize <= SHAPE_DIM4) << "The shape.size() only support 1~4";
-    std::vector<DataType> CUMSUM_SUPPORT_DATATYPES = {DataType::DT_FP32, DataType::DT_INT32, DataType::DT_INT16};
+    ASSERT(SHAPE_DIM1 <= shapeSize && shapeSize <= SHAPE_DIM5) << "The shape.size() only support 1~5";
+    std::vector<DataType> CUMSUM_SUPPORT_DATATYPES = {DataType::DT_FP32, DataType::DT_FP16, DataType::DT_INT32,
+        DataType::DT_INT16, DataType::DT_INT8, DataType::DT_BF16};
     ASSERT(std::find(CUMSUM_SUPPORT_DATATYPES.begin(), CUMSUM_SUPPORT_DATATYPES.end(), dataType) !=
            CUMSUM_SUPPORT_DATATYPES.end()) << "The datatype is not supported";
-    int tmpAxis = axis < 0 ? shapeSize + axis : axis;
+    int tmpAxis0 = axis < 0 ? shapeSize + axis : axis;
     bool flag = input.GetShape().size() == 1 ? true : false;
     if (flag) {
-        ASSERT(tmpAxis == 0) << "when input.GetShape().size() is 1, axis must be 0";
+        ASSERT(tmpAxis0 == 0) << "when input.GetShape().size() is 1, axis must be 0";
     }
-    ASSERT(tmpAxis == 0 || static_cast<size_t>(tmpAxis) < shapeSize) << "The tmpAxis should be 0 and less than shape size";
+    ASSERT(tmpAxis0 == 0 || static_cast<size_t>(tmpAxis0) < shapeSize) << "The tmpAxis0 should be 0 and less than shape size";
 
-    Tensor result(input.GetDataType(), input.GetShape());
-    CALL(CumSum, *Program::GetInstance().GetCurrentFunction(), {input.GetStorage(), result.GetStorage(), tmpAxis, flag});
-    result.GetStorage()->UpdateDynValidShape(input.GetStorage()->dynValidShape_);
-    return result;
+    const int n_1 = input.GetShape().size() - 1;
+    const int n_2 = input.GetShape().size() - 2;
+    if ((dataType != DataType::DT_INT8) && tmpAxis0 > 0 && tmpAxis0 == n_1) {
+        Tensor tmpInput = Transpose(input, {n_2, n_1});
+        const int transposedAxis = n_2;
+
+        VecTile tmpVectile = TileShape::Current().GetVecTile();
+        int64_t tmp = tmpVectile.tile[n_2];
+        tmpVectile.tile[n_2] = tmpVectile.tile[n_1];
+        tmpVectile.tile[n_1] = tmp;
+        TileShape::Current().SetVecTile(tmpVectile);
+
+        Tensor result(tmpInput.GetDataType(), tmpInput.GetShape());
+        CALL(CumSum, *Program::GetInstance().GetCurrentFunction(),
+            {tmpInput.GetStorage(), result.GetStorage(), transposedAxis, flag});
+
+        Tensor tmpresult = Transpose(result, {n_2, n_1});
+        tmpresult.GetStorage()->UpdateDynValidShape(input.GetStorage()->dynValidShape_);
+        return tmpresult;
+    } else {
+        Tensor result(input.GetDataType(), input.GetShape());
+        CALL(CumSum, *Program::GetInstance().GetCurrentFunction(),
+            {input.GetStorage(), result.GetStorage(), tmpAxis0, flag});
+        result.GetStorage()->UpdateDynValidShape(input.GetStorage()->dynValidShape_);
+        return result;
+    }
 }
 
 void CumSumOperationTileFunc(Function &function, const TileShape &tileShape,
diff --git a/framework/src/interface/tileop/a2a3/dynamic/vector_dyn.h b/framework/src/interface/tileop/a2a3/dynamic/vector_dyn.h
index b8441917d91e3b7418c897f141e7c03ac7055d7f..3f42f20aa0667df40b81d4bacf5f7ee636162b8d 100644
--- a/framework/src/interface/tileop/a2a3/dynamic/vector_dyn.h
+++ b/framework/src/interface/tileop/a2a3/dynamic/vector_dyn.h
@@ -15,6 +15,7 @@
 
 #include "tileop_common.h"
 #include "vector.h"
+#include <array>
 
 #include <type_traits>
 
@@ -2526,7 +2527,7 @@ TILEOP void DynTindexAdd(__ubuf__ T *dst, __ubuf__ T *src, __ubuf__ T1 *indices,
     wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
 }
 
-template <typename T>
+template <typename T, typename T1>
 TILEOP void CumSumPublicTool(
     __ubuf__ T *dst, __ubuf__ T *input, unsigned TShape3, uint64_t offset, uint32_t idx, uint64_t stride) {
     uint32_t rptElm = REPEAT_BYTE / sizeof(T);
@@ -2559,7 +2560,7 @@ TILEOP void CumSumPublicTool(
     }
 }
 
-template <typename T, unsigned axis>
+template <typename T, typename T1, unsigned axis>
 TILEOP void CumSumAxis0_2(__ubuf__ T *dst, __ubuf__ T *input, unsigned TShape0, unsigned TShape1, unsigned TShape2,
     unsigned TShape3, uint64_t inputStride1, uint64_t inputStride2, uint64_t inputStride3) {
     uint64_t offset = 0;
@@ -2568,47 +2569,179 @@ TILEOP void CumSumAxis0_2(__ubuf__ T *dst, __ubuf__ T *input, unsigned TShape0,
             for (uint32_t k = 0; k < TShape2; ++k) {
                 offset = i * inputStride1 + j * inputStride2 + k * inputStride3;
                 if constexpr (axis == 0) {
-                    CumSumPublicTool<T>(dst, input, TShape3, offset, i, inputStride1);
+                    CumSumPublicTool<T, T1>(dst, input, TShape3, offset, i, inputStride1);
                 } else if constexpr (axis == 1) {
-                    CumSumPublicTool<T>(dst, input, TShape3, offset, j, inputStride2);
+                    CumSumPublicTool<T, T1>(dst, input, TShape3, offset, j, inputStride2);
                 } else if constexpr (axis == 2) {
-                    CumSumPublicTool<T>(dst, input, TShape3, offset, k, inputStride3);
+                    CumSumPublicTool<T, T1>(dst, input, TShape3, offset, k, inputStride3);
                 }
             }
         }
     }
 }
 
-template <typename T, unsigned inputRawShape1, unsigned inputRawShape2, unsigned inputRawShape3, unsigned axis,
-    bool flag>
+template <typename T>
+TILEOP void CumSumAdd(__ubuf__ T *dst, __ubuf__ T *src0, __ubuf__ T *src1, int repeat, int dstBlockStride,
+    int src0BlockStride, int src1BlockStride, int dstRepeatStride, int src0RepeatStride, int src1RepeatStride) {
+    int n = repeat / REPEAT_MAX;
+    int rest = repeat % REPEAT_MAX;
+    constexpr int elePerRepeat = REPEAT_BYTE / sizeof(T);
+    __ubuf__ T *dst_ = dst;
+    __ubuf__ T *src0_ = src0;
+    __ubuf__ T *src1_ = src1;
+
+    for (size_t i = 0; i < n; i++) {
+        vadd(dst_, src0_, src1_, REPEAT_MAX, dstBlockStride, src0BlockStride, src1BlockStride, dstRepeatStride,
+            src0RepeatStride, src1RepeatStride);
+        dst_ += REPEAT_MAX * elePerRepeat;
+        src0_ += REPEAT_MAX * elePerRepeat;
+        src1_ += REPEAT_MAX * elePerRepeat;
+    }
+    vadd(dst_, src0_, src1_, rest, dstBlockStride, src0BlockStride, src1BlockStride, dstRepeatStride, src0RepeatStride,
+        src1RepeatStride);
+}
+
+template <typename T>
+TILEOP void CumSumAdds(__ubuf__ T *dst, __ubuf__ T *src0, T src1, int repeat, int dstBlockStride, int src0BlockStride,
+    int dstRepeatStride, int src0RepeatStride) {
+    int n = repeat / REPEAT_MAX;
+    int rest = repeat % REPEAT_MAX;
+    constexpr int elePerRepeat = REPEAT_BYTE / sizeof(T);
+
+    __ubuf__ T *dst_ = dst;
+    __ubuf__ T *src0_ = src0;
+
+    for (size_t i = 0; i < n; i++) {
+        vadds(dst_, src0_, src1, REPEAT_MAX, dstBlockStride, src0BlockStride, dstRepeatStride, src0RepeatStride);
+        dst_ += REPEAT_MAX * elePerRepeat;
+        src0_ += REPEAT_MAX * elePerRepeat;
+    }
+    vadds(dst_, src0_, src1, rest, dstBlockStride, src0BlockStride, dstRepeatStride, src0RepeatStride);
+}
+
+template <typename T, unsigned int RawShape0, unsigned int RawShape1, int Axis>
+TILEOP void CumSum2d(__ubuf__ T *dst, __ubuf__ T *src) {
+    if constexpr (Axis == 0) {
+        int repeat = RawShape1 * sizeof(T) / REPEAT_BYTE;
+        int rest = (RawShape1 * sizeof(T) % REPEAT_BYTE) / sizeof(T);
+        constexpr int blockStride = 1;
+        constexpr int repeatStride = 8;
+        constexpr int elePerRepeat = REPEAT_BYTE / sizeof(T);
+
+        if (repeat > 0) {
+            CumSumAdds(dst, src, (T)0, repeat, blockStride, blockStride, repeatStride, repeatStride);
+        }
+        if (rest > 0) {
+            SetContinuousMask(rest);
+            set_flag(PIPE_S, PIPE_V, EVENT_ID7);
+            wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
+            CumSumAdds(dst + repeat * elePerRepeat, src + repeat * elePerRepeat, (T)0, 1, blockStride, blockStride,
+                repeatStride, repeatStride);
+            set_flag(PIPE_V, PIPE_S, EVENT_ID7);
+            wait_flag(PIPE_V, PIPE_S, EVENT_ID7);
+            set_vector_mask(-1, -1);
+        }
+        pipe_barrier(PIPE_V);
+
+        int i = 1;
+        while (i < RawShape0) {
+            repeat = (RawShape0 - i) * RawShape1 * sizeof(T) / REPEAT_BYTE;
+            rest = ((RawShape0 - i) * RawShape1 * sizeof(T) % REPEAT_BYTE) / sizeof(T);
+            if (repeat > 0) {
+                CumSumAdd(dst + i * RawShape1, src, src + i * RawShape1, repeat, blockStride, blockStride, blockStride,
+                    repeatStride, repeatStride, repeatStride);
+            }
+            if (rest > 0) {
+                SetContinuousMask(rest);
+                set_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                CumSumAdd(dst + i * RawShape1 + repeat * elePerRepeat, src + repeat * elePerRepeat,
+                    src + i * RawShape1 + repeat * elePerRepeat, 1, blockStride, blockStride, blockStride, repeatStride,
+                    repeatStride, repeatStride);
+                set_flag(PIPE_V, PIPE_S, EVENT_ID7);
+                wait_flag(PIPE_V, PIPE_S, EVENT_ID7);
+                set_vector_mask(-1, -1);
+            }
+            pipe_barrier(PIPE_V);
+            if (repeat > 0) {
+                CumSumAdds(src + i * RawShape1, dst + i * RawShape1, (T)0, repeat, blockStride, blockStride,
+                    repeatStride, repeatStride);
+            }
+            if (rest > 0) {
+                SetContinuousMask(rest);
+                set_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                CumSumAdds(src + i * RawShape1 + repeat * elePerRepeat, dst + i * RawShape1 + repeat * elePerRepeat,
+                    (T)0, 1, blockStride, blockStride, repeatStride, repeatStride);
+                set_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
+                set_vector_mask(-1, -1);
+            }
+            pipe_barrier(PIPE_V);
+            i *= 2;
+        }
+    } else {
+        __ubuf__ T *dst_ = dst;
+        __ubuf__ T *src_ = src;
+        for (size_t i = 0; i < RawShape0; i++) {
+            dst_[0] = src_[0];
+            for (size_t j = 1; j < RawShape1; j++) {
+                if constexpr (std::is_same_v<T, half>) {
+                    T tmp = static_cast<float>(src_[j]) + static_cast<float>(dst_[j - 1]);
+                    dst_[j] = tmp;
+                } else {
+                    dst_[j] = dst_[j - 1] + src_[j];
+                }
+            }
+            dst_ += RawShape1;
+            src_ += RawShape1;
+        }
+    }
+}
+
+template <int axis, unsigned... RawShapes>
+[host, aicore] constexpr int product() {
+    constexpr size_t rank = sizeof...(RawShapes);
+    constexpr std::array<unsigned int, rank> dims = {RawShapes...};
+    static_assert(axis >= 0 && axis < rank, "Axis out of bounds");
+
+    int product = 1;
+    for (size_t i = axis; i < rank; ++i) {
+        product *= dims[i];
+    }
+    return product;
+}
+
+template <typename T, int Axis, unsigned... RawShapes>
+TILEOP void CumSum(__ubuf__ T *dst, __ubuf__ T *src) {
+    constexpr size_t rank = sizeof...(RawShapes);
+
+    if constexpr (Axis == 0) {
+        constexpr int dim0 = product<0, RawShapes...>() / product<1, RawShapes...>();
+        constexpr int dim1 = product<1, RawShapes...>();
+        CumSum2d<T, dim0, dim1, 0>(dst, src);
+    } else if constexpr (Axis == (int)rank - 1) {
+        constexpr int dim0 = product<0, RawShapes...>() / product<Axis, RawShapes...>();
+        constexpr int dim1 = product<Axis, RawShapes...>();
+        CumSum2d<T, dim0, dim1, 1>(dst, src);
+    } else {
+        constexpr int loop = product<0, RawShapes...>() / product<Axis, RawShapes...>();
+        constexpr int dim0 = product<Axis, RawShapes...>() / product<Axis + 1, RawShapes...>();
+        constexpr int dim1 = product<Axis + 1, RawShapes...>();
+        for (size_t i = 0; i < loop; i++) {
+            CumSum2d<T, dim0, dim1, 0>(dst + i * dim0 * dim1, src + i * dim0 * dim1);
+        }
+    }
+}
+
+template <typename T, unsigned inputRawShape0, unsigned inputRawShape1, unsigned inputRawShape2,
+    unsigned inputRawShape3, int axis, bool flag>
 TILEOP void DynTcumSum(
     __ubuf__ T *dst, __ubuf__ T *input, unsigned TShape0, unsigned TShape1, unsigned TShape2, unsigned TShape3) {
     set_flag(PIPE_V, PIPE_S, EVENT_ID7);
     wait_flag(PIPE_V, PIPE_S, EVENT_ID7);
-    uint64_t inputStride1 = inputRawShape1 * inputRawShape2 * inputRawShape3;
-    uint64_t inputStride2 = inputRawShape2 * inputRawShape3;
-    uint64_t inputStride3 = inputRawShape3;
 
-    if constexpr (axis != 3) {
-        CumSumAxis0_2<T, axis>(
-            dst, input, TShape0, TShape1, TShape2, TShape3, inputStride1, inputStride2, inputStride3);
-    } else {
-        uint64_t offset = 0;
-        for (uint32_t i = 0; i < TShape0; ++i) {
-            for (uint32_t j = 0; j < TShape1; ++j) {
-                for (uint32_t k = 0; k < TShape2; ++k) {
-                    for (uint32_t idx = 0; idx < TShape3; ++idx) {
-                        offset = i * inputStride1 + j * inputStride2 + k * inputStride3 + idx;
-                        if (idx == 0) {
-                            dst[offset] = input[offset];
-                        } else {
-                            dst[offset] = input[offset] + dst[offset - 1];
-                        }
-                    }
-                }
-            }
-        }
-    }
+    CumSum<T, axis, inputRawShape0, inputRawShape1, inputRawShape2, inputRawShape3>(dst, input);
 
     set_flag(PIPE_S, PIPE_V, EVENT_ID7);
     wait_flag(PIPE_S, PIPE_V, EVENT_ID7);
diff --git a/framework/tests/st/operation/python/vector_operator_golden.py b/framework/tests/st/operation/python/vector_operator_golden.py
index 6ba51591366f778470405c1684bdd18350605c69..add36240a86c61c39e3ecd55bedb501be36a0c5e 100644
--- a/framework/tests/st/operation/python/vector_operator_golden.py
+++ b/framework/tests/st/operation/python/vector_operator_golden.py
@@ -1409,8 +1409,14 @@ def gen_gatherelement_op_golden(
 def cumsum_golden_func(inputs: list, config: dict):
     params = config.get("params")
     axis = params["axis"]
-    inputTensor = torch.from_numpy(inputs[0])
+    if inputs[0].dtype == bfloat16:
+        inputTensor = torch.as_tensor(inputs[0].astype(np.float32)).to(torch.bfloat16)
+    else:
+        inputTensor = torch.from_numpy(inputs[0])
     res = torch.cumsum(inputTensor, axis)
+    if inputs[0].dtype == bfloat16:
+        res = res.to(torch.float32).numpy().astype(bfloat16)
+        return [res]
     
     return [res.numpy()]
 
