diff --git a/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.cpp b/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.cpp
index 3382a4eb1a7654d00981c7af7cb9dadbc81e8941..97eca70926f35544cdf6a54aeaff39ddfcf48f46 100644
--- a/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.cpp
+++ b/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.cpp
@@ -75,8 +75,8 @@ int OptimizeSort::GetNodePriority(std::unordered_map<Opcode, int> preNodePriorit
 }
 
 int OptimizeSort::GetMaxDepthSimple(Operation* op) {
-    auto it = depthCache.find(op);
-    if (it != depthCache.end()) {
+    auto it = depthCache_.find(op);
+    if (it != depthCache_.end()) {
         return it->second;
     }
 
@@ -86,7 +86,7 @@ int OptimizeSort::GetMaxDepthSimple(Operation* op) {
     }
 
     int depth = maxDepth + 1;
-    depthCache[op] = depth;
+    depthCache_[op] = depth;
     return depth;
 }
 
@@ -220,29 +220,31 @@ bool OptimizeSort::HasDependency(Operation* rollBackOp, Operation* backOp) {
 }
 
 // 在 curOpList 中将 advanceIndexList 中的序列提前到 rollBackIndex 之前,更新 curOpList
-void OptimizeSort::ReplaceIndex(std::vector<Operation*> &curOpList,
-    std::set<size_t> advanceIndexList, size_t rollBackIndex) {
+std::shared_ptr<std::vector<Operation*>> OptimizeSort::ReplaceIndex(std::shared_ptr<std::vector<Operation*>> curOpList,
+    std::set<size_t> &advanceIndexList, size_t rollBackIndex) {
     std::vector<Operation*> moveOpList;
     for (auto i : advanceIndexList) {
         APASS_LOG_DEBUG_F(Elements::Operation, "advance index: %d, op: %s",
-                i, GetOpInfo(curOpList[i]).c_str());
-        moveOpList.push_back(curOpList[i]);
+                i, GetOpInfo((*curOpList)[i]).c_str());
+        moveOpList.push_back((*curOpList)[i]);
     }
+    auto copyCurOpList = std::make_shared<std::vector<Operation* >>(*curOpList);
     for (auto it = advanceIndexList.rbegin(); it != advanceIndexList.rend(); ++it) {
-        curOpList.erase(curOpList.begin() + (*it));
+        copyCurOpList->erase(copyCurOpList->begin() + (*it));
     }
-    curOpList.insert(curOpList.begin() + rollBackIndex, moveOpList.begin(), moveOpList.end());
+    copyCurOpList->insert(copyCurOpList->begin() + rollBackIndex, moveOpList.begin(), moveOpList.end());
+    return copyCurOpList;
 }
 
-void OptimizeSort::GetPreNode(size_t i, std::vector<Operation*> curOpList, size_t rollBackIndex,
+void OptimizeSort::GetPreNode(size_t i, std::shared_ptr<std::vector<Operation*>> curOpList, size_t rollBackIndex,
     size_t backTraceIndex, std::set<size_t> &dependencyIndexList) {
     dependencyIndexList.insert(i);
     APASS_LOG_DEBUG_F(Elements::Operation, "dependencyIndexList push index: %d, Op: %s",
-        i, GetOpInfo(curOpList[i]).c_str());
-    for (auto preOp : inGraph[curOpList[i]]) {
-        auto it = std::find(curOpList.begin() + rollBackIndex + 1, curOpList.begin() + backTraceIndex, preOp);
-        if (it != curOpList.begin() + backTraceIndex) {
-            auto index = std::distance(curOpList.begin(), it);
+        i, GetOpInfo((*curOpList)[i]).c_str());
+    for (auto preOp : inGraph[(*curOpList)[i]]) {
+        auto it = std::find(curOpList->begin() + rollBackIndex + 1, curOpList->begin() + backTraceIndex, preOp);
+        if (it != curOpList->begin() + backTraceIndex) {
+            auto index = std::distance(curOpList->begin(), it);
             GetPreNode(index, curOpList, rollBackIndex, backTraceIndex, dependencyIndexList);
         }
     }
@@ -250,10 +252,10 @@ void OptimizeSort::GetPreNode(size_t i, std::vector<Operation*> curOpList, size_
 
 // 记录 curOpList 中从 rollBackIndex 到 backTraceIndex 中所有和 rollBack 没有后继依赖的点
 void OptimizeSort::GetListToAdvance(size_t rollBackIndex, size_t backTraceIndex,
-    std::vector<Operation*> curOpList, std::set<size_t> &advanceIndexList) {
+    std::shared_ptr<std::vector<Operation*>> curOpList, std::set<size_t> &advanceIndexList) {
     std::set<size_t> dependencyIndexList;
     for (size_t i = rollBackIndex + 1; i <= backTraceIndex; i++) {
-        if (HasDependency(curOpList[rollBackIndex], curOpList[i])) {
+        if (HasDependency((*curOpList)[rollBackIndex], (*curOpList)[i])) {
             GetPreNode(i, curOpList, rollBackIndex, backTraceIndex, dependencyIndexList);
         }
     }
@@ -261,49 +263,49 @@ void OptimizeSort::GetListToAdvance(size_t rollBackIndex, size_t backTraceIndex,
         if (dependencyIndexList.count(i) == 0) {
             advanceIndexList.insert(i);
             APASS_LOG_DEBUG_F(Elements::Operation, "advanceIndexList push index: %d, op: %s",
-                i, GetOpInfo(curOpList[i]).c_str());
+                i, GetOpInfo((*curOpList)[i]).c_str());
         }
     }
 }
 
 // rollBackIndex 位置回退
 Status OptimizeSort::RollBack(size_t &startIndex,
-    std::vector<Operation*> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap) {
+    std::shared_ptr<std::vector<Operation*>> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap) {
     APASS_LOG_DEBUG_F(Elements::Operation, "=====> Start RollBack.");
-    curOpList = backTraceOpList[backTraceOp].second;
-    MemoryType memType = recordOpBuffer[backTraceOp];
-    size_t backTraceIndex = backTraceOpList[backTraceOp].first + 1;
-    backTraceOp = curOpList[backTraceIndex];
+    curOpList = backTraceOpList_[backTraceOp_].second;
+    MemoryType memType = recordOpBuffer_[backTraceOp_];
+    size_t backTraceIndex = backTraceOpList_[backTraceOp_].first + 1;
+    backTraceOp_ = (*curOpList)[backTraceIndex];
     size_t rollBackIndex = backTraceIndex;
-    APASS_LOG_DEBUG_F(Elements::Operation, "backTraceOp: %s, backTraceIndex: %d, memType: %d",
-        GetOpInfo(backTraceOp).c_str(), backTraceIndex, memType);
+    APASS_LOG_DEBUG_F(Elements::Operation, "backTraceOp_: %s, backTraceIndex: %d, memType: %d",
+        GetOpInfo(backTraceOp_).c_str(), backTraceIndex, memType);
     std::set<size_t> advanceIndexList;
-    while (rollBackIndex < curOpList.size() && rollBackIndex > 0) {
+    while (rollBackIndex < curOpList->size() && rollBackIndex > 0) {
         rollBackIndex--;
-        Operation* rollBackOp = curOpList[rollBackIndex];
-        if (recordOpBuffer[rollBackOp] != memType || !(IsOpAlloc(rollBackOp)) || HasDependency(rollBackOp, backTraceOp)) {
+        Operation* rollBackOp = (*curOpList)[rollBackIndex];
+        if (recordOpBuffer_[rollBackOp] != memType || !(IsOpAlloc(rollBackOp)) || HasDependency(rollBackOp, backTraceOp_)) {
             continue;
         }
-        rollBackNodeOp = rollBackOp;
+        rollBackNodeOp_ = rollBackOp;
         APASS_LOG_DEBUG_F(Elements::Operation, "Select rollBackOp: %s, rollBackIndex: %d",
             GetOpInfo(rollBackOp).c_str(), rollBackIndex);
-        recordBufferAllocate = backTraceBufferAllocate;
-        recordOpList = backTraceOpList;
-        recordBufRefCount = backTraceBufRefCount;
+        recordBufferAllocate_ = backTraceBufferAllocate_;
+        recordOpList_ = backTraceOpList_;
+        recordBufRefCount_ = backTraceBufRefCount_;
         advanceIndexList.clear();
         GetListToAdvance(rollBackIndex, backTraceIndex, curOpList, advanceIndexList);
-        ReplaceIndex(curOpList, advanceIndexList, rollBackIndex);
+        curOpList = ReplaceIndex(curOpList, advanceIndexList, rollBackIndex);
         startIndex = rollBackIndex;
         APASS_LOG_DEBUG_F(Elements::Operation, "RollBack==>change startIndex: %d", startIndex);
         if (rollBackIndex != 0) {
-            curMemoryMap = recordBufferAllocate[curOpList[rollBackIndex-1]];
+            curMemoryMap = recordBufferAllocate_[(*curOpList)[rollBackIndex-1]];
             RecoverSymbol(startIndex - 1, curOpList);
             return SUCCESS;
         }
         curMemoryMap = {{MemoryType::MEM_L0A, 0}, {MemoryType::MEM_L0B, 0}, {MemoryType::MEM_L0C, 0}};
-        operations = curOpList;
-        for (auto op : curOpList) {
-            visitedOp[op] = false;
+        operations = *curOpList;
+        for (auto op : (*curOpList)) {
+            visitedOp_[op] = false;
         }
         InitBufRefCount();
         return SUCCESS;
@@ -313,25 +315,27 @@ Status OptimizeSort::RollBack(size_t &startIndex,
 }
 
 // 在 curOpList 中将 preOpList 中的序列提前到 startIndex 之后，更新 curOpList
-void OptimizeSort::ReorderOp(std::vector<size_t> &preIdx, std::vector<Operation*> &curOpList,
+std::shared_ptr<std::vector<Operation*>> OptimizeSort::ReorderOp(std::vector<size_t> &preIdx, std::shared_ptr<std::vector<Operation*>> curOpList,
     size_t startIndex) {
     // 对 perOpList 排序，再进行插入
     std::sort(preIdx.begin(), preIdx.end());
     std::vector<Operation*> moveOpList;
     APASS_LOG_DEBUG_F(Elements::Operation, "current index: %d, preIdx size: %d", startIndex, preIdx.size());
     for (auto i : preIdx) {
-        APASS_LOG_DEBUG_F(Elements::Operation, "preidx : %d, curOp: %s", i, GetOpInfo(curOpList[i]).c_str());
-        moveOpList.push_back(curOpList[i]);
+        APASS_LOG_DEBUG_F(Elements::Operation, "preidx : %d, curOp: %s", i, GetOpInfo((*curOpList)[i]).c_str());
+        moveOpList.push_back((*curOpList)[i]);
     }
+    auto copyCurOpList = std::make_shared<std::vector<Operation*>>(*curOpList);
     for (auto it = preIdx.rbegin(); it != preIdx.rend(); ++it) {
-        curOpList.erase(curOpList.begin() + (*it));
+        copyCurOpList->erase(copyCurOpList->begin() + (*it));
     }
-    curOpList.insert(curOpList.begin() + startIndex + 1, moveOpList.begin(), moveOpList.end());
+    copyCurOpList->insert(copyCurOpList->begin() + startIndex + 1, moveOpList.begin(), moveOpList.end());
+    return copyCurOpList;
 }
 
-void OptimizeSort::FindIndex(Operation* op, std::vector<Operation*> curOpList, size_t &index) {
-    for (size_t i = 0; i < curOpList.size(); i++) {
-        if (curOpList[i] == op) {
+void OptimizeSort::FindIndex(Operation* op, std::shared_ptr<std::vector<Operation*>> curOpList, size_t &index) {
+    for (size_t i = 0; i < curOpList->size(); i++) {
+        if ((*curOpList)[i] == op) {
             index = i;
             return;
         }
@@ -339,20 +343,20 @@ void OptimizeSort::FindIndex(Operation* op, std::vector<Operation*> curOpList, s
 }
 
 // 在curOpList中，向前遍历找到consumerIndex的前序未被访问的节点，并放入preOpList中
-Status OptimizeSort::FindConsumerList(size_t consumerIndex, std::vector<size_t> &preOpList, std::vector<Operation*> &curOpList) {
-    if (curOpList[consumerIndex] == backTraceOp) {
-        APASS_LOG_WARN_F(Elements::Operation, "backTraceOp is one of the predecessor node.");
+Status OptimizeSort::FindConsumerList(size_t consumerIndex, std::vector<size_t> &preOpList, std::shared_ptr<std::vector<Operation*>> curOpList) {
+    if ((*curOpList)[consumerIndex] == backTraceOp_) {
+        APASS_LOG_WARN_F(Elements::Operation, "backTraceOp_ is one of the predecessor node.");
         return FAILED;
     }
-    if (curOpList[consumerIndex] == rollBackNodeOp) {
-        APASS_LOG_WARN_F(Elements::Operation, "rollBackNodeOp is one of the predecessor node.");
+    if ((*curOpList)[consumerIndex] == rollBackNodeOp_) {
+        APASS_LOG_WARN_F(Elements::Operation, "rollBackNodeOp_ is one of the predecessor node.");
         return FAILED;
     }
-    visitedOp[curOpList[consumerIndex]] = true;
+    visitedOp_[(*curOpList)[consumerIndex]] = true;
     preOpList.push_back(consumerIndex);
-    APASS_LOG_DEBUG_F(Elements::Operation, "unvisited consumer idx: %d, op: %s", consumerIndex, GetOpInfo(curOpList[consumerIndex]).c_str());
-    for (auto op : inGraph[curOpList[consumerIndex]]) {
-        if (visitedOp[op] == false) {
+    APASS_LOG_DEBUG_F(Elements::Operation, "unvisited consumer idx: %d, op: %s", consumerIndex, GetOpInfo((*curOpList)[consumerIndex]).c_str());
+    for (auto op : inGraph[(*curOpList)[consumerIndex]]) {
+        if (visitedOp_[op] == false) {
             size_t index;
             FindIndex(op, curOpList, index);
             APASS_LOG_DEBUG_F(Elements::Operation, "consumer preIdx: %d, op: %s", index, GetOpInfo(op).c_str());
@@ -366,13 +370,13 @@ Status OptimizeSort::FindConsumerList(size_t consumerIndex, std::vector<size_t>
 }
 
 // 将 consumersGroup 和其前序依赖按原有顺序放入 preOpList
-Status OptimizeSort::UpdateOOperandPreDependence(size_t startIndex, std::vector<Operation*> &curOpList,
+Status OptimizeSort::UpdateOOperandPreDependence(size_t startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
     std::vector<Operation*> consumersGroup) {
     // curOpList 中向后找
     std::vector<size_t> preOpList;
     size_t index = startIndex;
-    while (index < curOpList.size()) {
-        if (std::find(consumersGroup.begin(), consumersGroup.end(), curOpList[index]) != consumersGroup.end()) {
+    while (index < curOpList->size()) {
+        if (std::find(consumersGroup.begin(), consumersGroup.end(), (*curOpList)[index]) != consumersGroup.end()) {
             APASS_LOG_DEBUG_F(Elements::Operation, "consumer Idx: %d", index);
             if (FindConsumerList(index, preOpList, curOpList) != SUCCESS) {
                 APASS_LOG_WARN_F(Elements::Operation, "FindConsumerList failed");
@@ -381,20 +385,20 @@ Status OptimizeSort::UpdateOOperandPreDependence(size_t startIndex, std::vector<
         }
         index++;
     }
-    ReorderOp(preOpList, curOpList, startIndex);
+    curOpList = ReorderOp(preOpList, curOpList, startIndex);
     return SUCCESS;
 }
 
-// 回溯后，将队列后面 op 的 visitedOp 状态还原回 false，并对应修改 refcount
-void OptimizeSort::RecoverSymbol(size_t startIndex, std::vector<Operation*> curOpList) {
-    APASS_LOG_DEBUG_F(Elements::Operation, "RecoverSymbol  startIdx: %d, curOp: %s", startIndex, GetOpInfo(curOpList[startIndex]).c_str());
-    bufRefCount = recordBufRefCount[curOpList[startIndex]];
-    for (size_t i = 0; i < curOpList.size(); i++) {
+// 回溯后，将队列后面 op 的 visitedOp_ 状态还原回 false，并对应修改 refcount
+void OptimizeSort::RecoverSymbol(size_t startIndex, std::shared_ptr<std::vector<Operation*>> curOpList) {
+    APASS_LOG_DEBUG_F(Elements::Operation, "RecoverSymbol  startIdx: %d, curOp: %s", startIndex, GetOpInfo((*curOpList)[startIndex]).c_str());
+    bufRefCount = recordBufRefCount_[(*curOpList)[startIndex]];
+    for (size_t i = 0; i < curOpList->size(); i++) {
         if (i > startIndex) {
-            visitedOp[curOpList[i]] = false;
+            visitedOp_[(*curOpList)[i]] = false;
             continue;
         }
-        visitedOp[curOpList[i]] = true;
+        visitedOp_[(*curOpList)[i]] = true;
     }
 }
 
@@ -402,35 +406,35 @@ void OptimizeSort::RecoverSymbol(size_t startIndex, std::vector<Operation*> curO
 void OptimizeSort::GetConsumerGroup(std::unordered_set<Operation*> &consumers, std::vector<Operation*> &consumersGroup) {
     for (auto op : consumers) {
         APASS_LOG_DEBUG_F(Elements::Operation, "consumer: %s", GetOpInfo(op).c_str());
-        if (!visitedOp[op]) {
+        if (!visitedOp_[op]) {
             consumersGroup.push_back(op);
             APASS_LOG_DEBUG_F(Elements::Operation, "unvisited consumer: %s", GetOpInfo(op).c_str());
         }
     }
 }
 
-void OptimizeSort::GetStackTop(size_t &startIndex, std::vector<Operation*> &curOpList,
+void OptimizeSort::GetStackTop(size_t &startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
     std::map<MemoryType, int64_t> &curMemoryMap) {
-    auto topNode = needFreeOpStack.top();
-    needFreeOpStack.pop();
-    curOpList = recordOpList[topNode.first].second;
-    startIndex = recordOpList[topNode.first].first;
-    curMemoryMap = recordBufferAllocate[topNode.first];
+    auto topNode = needFreeOpStack_.top();
+    needFreeOpStack_.pop();
+    curOpList = recordOpList_[topNode.first].second;
+    startIndex = recordOpList_[topNode.first].first;
+    curMemoryMap = recordBufferAllocate_[topNode.first];
 }
 
 Status OptimizeSort::BacktraceOnMemoryExceeded(size_t &startIndex,
-    std::vector<Operation*> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap) {
+    std::shared_ptr<std::vector<Operation*>> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap) {
     APASS_LOG_DEBUG_F(Elements::Tensor, "=====> Start Backtrace.");
-    MemoryType memType = curOpList[startIndex]->GetOutputOperand(0)->GetMemoryTypeOriginal();
+    MemoryType memType = (*curOpList)[startIndex]->GetOutputOperand(0)->GetMemoryTypeOriginal();
     std::vector<Operation*> consumersGroup;
-    while (startIndex < curOpList.size() && startIndex > 0) {
+    while (startIndex < curOpList->size() && startIndex > 0) {
         startIndex--;
-        auto op = curOpList[startIndex];
-        if (!needFreeOpStack.empty() && needFreeOpStack.top().first == curOpList[startIndex]) {
-            APASS_LOG_DEBUG_F(Elements::Operation, "Having traversed %s, the stack needs to be popped", GetOpInfo(curOpList[startIndex]).c_str());
+        auto op = (*curOpList)[startIndex];
+        if (!needFreeOpStack_.empty() && needFreeOpStack_.top().first == (*curOpList)[startIndex]) {
+            APASS_LOG_DEBUG_F(Elements::Operation, "Having traversed %s, the stack needs to be popped", GetOpInfo((*curOpList)[startIndex]).c_str());
             break;
         }
-        if (recordOpBuffer[op] != memType || IsOpAlloc(op)) {
+        if (recordOpBuffer_[op] != memType || IsOpAlloc(op)) {
             continue;
         }
         APASS_LOG_DEBUG_F(Elements::Operation, "===>start to find unvisited consumer");
@@ -443,10 +447,10 @@ Status OptimizeSort::BacktraceOnMemoryExceeded(size_t &startIndex,
         RecoverSymbol(startIndex, curOpList);
         GetConsumerGroup(outGraph[op], consumersGroup);
         APASS_LOG_DEBUG_F(Elements::Operation, "push %s to stack", GetOpInfo(op).c_str());
-        curMemoryMap = recordBufferAllocate[op];
-        needFreeOpStack.push(make_pair(op, recordOpBuffer[op]));
+        curMemoryMap = recordBufferAllocate_[op];
+        needFreeOpStack_.push(make_pair(op, recordOpBuffer_[op]));
         if (UpdateOOperandPreDependence(startIndex, curOpList, consumersGroup) != SUCCESS) {
-            needFreeOpStack.pop();
+            needFreeOpStack_.pop();
             APASS_LOG_DEBUG_F(Elements::Operation, "UpdateOOperandPreDependence failed.");
             continue;
         }
@@ -454,13 +458,13 @@ Status OptimizeSort::BacktraceOnMemoryExceeded(size_t &startIndex,
         APASS_LOG_DEBUG_F(Elements::Operation, "Backtrace==>change startIndex: %d", startIndex);
         return SUCCESS;
     }
-    if (needFreeOpStack.empty()) {
+    if (needFreeOpStack_.empty()) {
         APASS_LOG_WARN_F(Elements::Tensor, "Stack is empty. Start to rollback.");
         return FAILED;
     }
     GetStackTop(startIndex, curOpList, curMemoryMap);
     RecoverSymbol(startIndex, curOpList);
-    APASS_LOG_DEBUG_F(Elements::Operation, "pop %s from stack", GetOpInfo(curOpList[startIndex]).c_str());
+    APASS_LOG_DEBUG_F(Elements::Operation, "pop %s from stack", GetOpInfo((*curOpList)[startIndex]).c_str());
     if (BacktraceOnMemoryExceeded(startIndex, curOpList, curMemoryMap) != SUCCESS) {
         APASS_LOG_WARN_F(Elements::Tensor, "BacktraceOnMemoryExceeded Failed");
         return FAILED;
@@ -525,26 +529,26 @@ Status OptimizeSort::RetireOpBuffer(std::map<MemoryType, int64_t> &curMemoryMap,
     return SUCCESS;
 }
 
-void OptimizeSort::OpMemoryUpdate(Operation* op, size_t startIndex, std::vector<Operation*> curOpList,
+void OptimizeSort::OpMemoryUpdate(Operation* op, size_t startIndex, std::shared_ptr<std::vector<Operation*>> curOpList,
     std::map<MemoryType, int64_t> curMemoryMap) {
-    recordOpList[op] = make_pair(startIndex, curOpList);
-    recordBufferAllocate[op] = curMemoryMap;
-    recordOpBuffer[op] = op->GetOutputOperand(0)->GetMemoryTypeOriginal();
-    recordBufRefCount[op] = bufRefCount;
+    recordOpList_[op] = make_pair(startIndex, curOpList);
+    recordBufferAllocate_[op] = curMemoryMap;
+    recordOpBuffer_[op] = op->GetOutputOperand(0)->GetMemoryTypeOriginal();
+    recordBufRefCount_[op] = bufRefCount;
 }
 
-Status OptimizeSort::AllocExecute(Operation* op, std::vector<Operation*> &curOpList,
+Status OptimizeSort::AllocExecute(Operation* op, std::shared_ptr<std::vector<Operation*>> &curOpList,
     std::map<MemoryType, int64_t> &curMemoryMap, size_t &startIndex, bool &isContinue) {
     APASS_LOG_DEBUG_F(Elements::Operation, "alloc op: %s", GetOpInfo(op).c_str());
     auto tensor = op->GetOutputOperand(0);
     if (IsBufferFull(curMemoryMap, tensor->GetMemoryTypeOriginal(), ShapeCeilAlign(tensor->GetShape(), tensor->Datatype()))) {
         APASS_LOG_DEBUG_F(Elements::Operation, "The memory of %s needs to be released", std::to_string(tensor->GetMemoryTypeOriginal()).c_str());
-        backTraceOp = curOpList[startIndex];
-        backTraceBufferAllocate = recordBufferAllocate;
-        backTraceOpList = recordOpList;
-        backTraceBufRefCount = recordBufRefCount;
-        APASS_LOG_DEBUG_F(Elements::Operation, "backTraceOp: %s, backTraceIndex: %d, memType: %d",
-            GetOpInfo(backTraceOp).c_str(), backTraceOpList[backTraceOp].first, recordOpBuffer[backTraceOp]);
+        backTraceOp_ = (*curOpList)[startIndex];
+        backTraceBufferAllocate_ = recordBufferAllocate_;
+        backTraceOpList_ = recordOpList_;
+        backTraceBufRefCount_ = recordBufRefCount_;
+        APASS_LOG_DEBUG_F(Elements::Operation, "backTraceOp_: %s, backTraceIndex: %d, memType: %d",
+            GetOpInfo(backTraceOp_).c_str(), backTraceOpList_[backTraceOp_].first, recordOpBuffer_[backTraceOp_]);
         APASS_LOG_DEBUG_F(Elements::Operation, "=====> Need backtrace.");
         if (BacktraceOnMemoryExceeded(startIndex, curOpList, curMemoryMap) != SUCCESS) {
             if (RollBack(startIndex, curOpList, curMemoryMap) != SUCCESS) {
@@ -560,14 +564,14 @@ Status OptimizeSort::AllocExecute(Operation* op, std::vector<Operation*> &curOpL
     return SUCCESS;
 }
 
-Status OptimizeSort::OpListExecute(std::vector<Operation*> &curOpList,
+Status OptimizeSort::OpListExecute(std::shared_ptr<std::vector<Operation*>> &curOpList,
     std::map<MemoryType, int64_t> &curMemoryMap, size_t &startIndex) {
     APASS_LOG_DEBUG_F(Elements::Operation, "===>Start opListExecute, startIndex: %d", startIndex);
-    if (curOpList.empty()) {
-        curOpList = operations;
+    if (curOpList->empty()) {
+        curOpList = std::make_shared<std::vector<Operation*>>(operations);
     }
-    while (startIndex < curOpList.size()) {
-        auto op = curOpList[startIndex];
+    while (startIndex < curOpList->size()) {
+        auto op = (*curOpList)[startIndex];
         OpMemoryUpdate(op, startIndex, curOpList, curMemoryMap);
         APASS_LOG_DEBUG_F(Elements::Operation, "execute op: %s, index: %d", GetOpInfo(op).c_str(), startIndex);
         if (IsOpAlloc(op)) {
@@ -584,9 +588,9 @@ Status OptimizeSort::OpListExecute(std::vector<Operation*> &curOpList,
                 APASS_LOG_ERROR_F(Elements::Tensor, "Allocate tensor[%u] failed.", tensor->GetMagic());
                 return FAILED;
             }
+            OpMemoryUpdate(op, startIndex, curOpList, curMemoryMap);
         }
-        visitedOp[op] = true;
-        OpMemoryUpdate(op, startIndex, curOpList, curMemoryMap);
+        visitedOp_[op] = true;
         if (RetireOpBuffer(curMemoryMap, op) != SUCCESS) {
             APASS_LOG_ERROR_F(Elements::Operation, "RetireOp failed! %s", GetOpInfo(op).c_str());
             return FAILED;
@@ -594,25 +598,25 @@ Status OptimizeSort::OpListExecute(std::vector<Operation*> &curOpList,
         OpMemoryUpdate(op, startIndex, curOpList, curMemoryMap);
         startIndex += 1;
     }
-    opFinish = true;
+    opFinish_ = true;
     return SUCCESS;
 }
 
 Status OptimizeSort::ExecuteOp() {
-    std::vector<Operation*> curOpList;
+    auto curOpList = std::make_shared<std::vector<Operation*>>();
     std::map<MemoryType, int64_t> curMemoryMap = {{MemoryType::MEM_L0A, 0}, {MemoryType::MEM_L0B, 0},
         {MemoryType::MEM_L0C, 0}};
     size_t startIndex{0};
     for (auto &op : operations) {
-        visitedOp[op] = false;
+        visitedOp_[op] = false;
     }
-    while(!opFinish) {
+    while(!opFinish_) {
         if (OpListExecute(curOpList, curMemoryMap, startIndex) != SUCCESS) {
             APASS_LOG_ERROR_F(Elements::Operation, "OpListExecute failed.");
             return FAILED;
         }
     }
-    operations = curOpList;
+    operations = *curOpList;
     return SUCCESS;
 }
 
diff --git a/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.h b/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.h
index f4da785ca45dcb32948f6b2915a9568eddd77129..87a040c0a8fd5e1d265a7930b2fea7625468ae52 100644
--- a/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.h
+++ b/framework/src/passes/block_graph_pass/schedule_ooo/optimize_sort.h
@@ -28,22 +28,22 @@ public:
     std::vector<Operation*> operations;
     Function &function_;
 
-    bool opFinish{false};
-    std::map<Operation*, std::map<MemoryType, int64_t>> recordBufferAllocate;
-    std::map<Operation*, std::pair<size_t, std::vector<Operation*>>> recordOpList;
-    std::map<Operation*, MemoryType> recordOpBuffer;
-    std::stack<std::pair<Operation*, MemoryType>> needFreeOpStack;
-    std::map<Operation*, bool> visitedOp;
-    std::map<Operation*, std::unordered_map<int, int>> recordBufRefCount;
+    bool opFinish_{false};
+    std::map<Operation*, std::map<MemoryType, int64_t>> recordBufferAllocate_;
+    std::map<Operation*, std::pair<size_t, std::shared_ptr<std::vector<Operation*>>>> recordOpList_;
+    std::map<Operation*, MemoryType> recordOpBuffer_;
+    std::stack<std::pair<Operation*, MemoryType>> needFreeOpStack_;
+    std::map<Operation*, bool> visitedOp_;
+    std::map<Operation*, std::unordered_map<int, int>> recordBufRefCount_;
 
     // 回溯点位置,当前执行op的全部信息,用于后期回退
-    Operation* backTraceOp{nullptr};
-    std::map<Operation*, std::map<MemoryType, int64_t>> backTraceBufferAllocate;
-    std::map<Operation*, std::pair<size_t, std::vector<Operation*>>> backTraceOpList;
-    std::map<Operation*, std::unordered_map<int, int>> backTraceBufRefCount;
+    Operation* backTraceOp_{nullptr};
+    std::map<Operation*, std::map<MemoryType, int64_t>> backTraceBufferAllocate_;
+    std::map<Operation*, std::pair<size_t, std::shared_ptr<std::vector<Operation*>>>> backTraceOpList_;
+    std::map<Operation*, std::unordered_map<int, int>> backTraceBufRefCount_;
     // 回退点,防止死循环
-    Operation* rollBackNodeOp{nullptr};
-    std::unordered_map<Operation*, int> depthCache;
+    Operation* rollBackNodeOp_{nullptr};
+    std::unordered_map<Operation*, int> depthCache_;
 
     void opListInit();
     Status SortOps();
@@ -64,36 +64,36 @@ public:
     void UpdatePreNodeQueue(std::unordered_set<Operation*> &curr, std::unordered_set<Operation*> &preNodeTotal,
         std::map<Operation*, bool>& visited);
 
-    void ReorderOp(std::vector<size_t> &preIdx, std::vector<Operation*> &curOpList, size_t startIndex);
-    void FindIndex(Operation* op, std::vector<Operation*> curOpList, size_t &index);
-    Status FindConsumerList(size_t consumerIndex, std::vector<size_t> &preOpList, std::vector<Operation*> &curOpList);
-    Status UpdateOOperandPreDependence(size_t startIndex, std::vector<Operation*> &curOpList,
+    std::shared_ptr<std::vector<Operation*>> ReorderOp(std::vector<size_t> &preIdx, std::shared_ptr<std::vector<Operation*>> curOpList, size_t startIndex);
+    void FindIndex(Operation* op, std::shared_ptr<std::vector<Operation*>> curOpList, size_t &index);
+    Status FindConsumerList(size_t consumerIndex, std::vector<size_t> &preOpList, std::shared_ptr<std::vector<Operation*>> curOpList);
+    Status UpdateOOperandPreDependence(size_t startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
         std::vector<Operation*> consumersGroup);
-    void RecoverSymbol(size_t startIndex, std::vector<Operation*> curOpList);
+    void RecoverSymbol(size_t startIndex, std::shared_ptr<std::vector<Operation*>> curOpList);
     void GetConsumerGroup(std::unordered_set<Operation*> &consumers, std::vector<Operation*> &consumersGroup);
-    void GetStackTop(size_t &startIndex, std::vector<Operation*> &curOpList,
+    void GetStackTop(size_t &startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
         std::map<MemoryType, int64_t> &curMemoryMap);
-    Status BacktraceOnMemoryExceeded(size_t &startIndex, std::vector<Operation*> &curOpList,
+    Status BacktraceOnMemoryExceeded(size_t &startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
         std::map<MemoryType, int64_t> &curMemoryMap);
     bool IsBufferFull(std::map<MemoryType, int64_t> curMemoryMap, MemoryType memType, int64_t size);
     Status ModifyBuffer(std::map<MemoryType, int64_t> &curMemoryMap, MemoryType memType, int64_t size, bool isAdd);
     Status RetireOpBuffer(std::map<MemoryType, int64_t> &curMemoryMap, Operation* op);
-    void OpMemoryUpdate(Operation* op, size_t startIndex, std::vector<Operation*> curOpList,
+    void OpMemoryUpdate(Operation* op, size_t startIndex, std::shared_ptr<std::vector<Operation*>> curOpList,
         std::map<MemoryType, int64_t> curMemoryMap);
-    Status AllocExecute(Operation* op, std::vector<Operation*> &curOpList,
+    Status AllocExecute(Operation* op, std::shared_ptr<std::vector<Operation*>> &curOpList,
         std::map<MemoryType, int64_t> &curMemoryMap, size_t &startIndex, bool &isContinue);
-    Status OpListExecute(std::vector<Operation*> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap,
+    Status OpListExecute(std::shared_ptr<std::vector<Operation*>> &curOpList, std::map<MemoryType, int64_t> &curMemoryMap,
         size_t &startIndex);
     Status ExecuteOp();
     void AllocAhead();
 
-    void ReplaceIndex(std::vector<Operation*> &curOpList, std::set<size_t> advanceIndexList, size_t rollBackIndex);
+    std::shared_ptr<std::vector<Operation*>> ReplaceIndex(std::shared_ptr<std::vector<Operation*>> curOpList, std::set<size_t> &advanceIndexList, size_t rollBackIndex);
     bool HasDependency(Operation* rollBackOp, Operation* backOp);
-    void GetPreNode(size_t i, std::vector<Operation*> curOpList, size_t rollBackIndex,
+    void GetPreNode(size_t i, std::shared_ptr<std::vector<Operation*>> curOpList, size_t rollBackIndex,
         size_t backTraceIndex, std::set<size_t> &dependencyIndexList);
     void GetListToAdvance(size_t rollBackIndex, size_t backTraceIndex,
-        std::vector<Operation*> curOpList, std::set<size_t> &advanceIndexList);
-    Status RollBack(size_t &startIndex, std::vector<Operation*> &curOpList,
+        std::shared_ptr<std::vector<Operation*>> curOpList, std::set<size_t> &advanceIndexList);
+    Status RollBack(size_t &startIndex, std::shared_ptr<std::vector<Operation*>> &curOpList,
         std::map<MemoryType, int64_t> &curMemoryMap);
 
 
